### PR TITLE
Enable all warnings

### PR DESCRIPTION
**Pull request by raulriera**
## What

Taken a page out of the Webkit book, turning all warnings to yes.

### CODE REVIEW

File: FunctionalTableData.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -567,26 +567,35 @@
...
 				CLANG_ENABLE_MODULES = YES;
 				CLANG_ENABLE_OBJC_ARC = YES;
+				CLANG_WARN_ASSIGN_ENUM = YES;
 				CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING = YES;
 				CLANG_WARN_BOOL_CONVERSION = YES;
 				CLANG_WARN_COMMA = YES;
 				CLANG_WARN_CONSTANT_CONVERSION = YES;
+				CLANG_WARN_CXX0X_EXTENSIONS = YES;
 				CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS = YES;
-				CLANG_WARN_DIRECT_OBJC_ISA_USAGE = YES_ERROR;
```
Comment (krbarnes) [inline FunctionalTableData.xcodeproj/project.pbxproj]: did you mean to downgrade this one?

File: FunctionalTableData.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -567,26 +567,35 @@
...
 				CLANG_WARN_ENUM_CONVERSION = YES;
+				CLANG_WARN_IMPLICIT_SIGN_CONVERSION = YES;
 				CLANG_WARN_INFINITE_RECURSION = YES;
 				CLANG_WARN_INT_CONVERSION = YES;
 				CLANG_WARN_NON_LITERAL_NULL_CONVERSION = YES;
+				CLANG_WARN_OBJC_EXPLICIT_OWNERSHIP_TYPE = YES;
+				CLANG_WARN_OBJC_IMPLICIT_ATOMIC_PROPERTIES = YES;
 				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
 				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
-				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
```
Comment (krbarnes) [inline FunctionalTableData.xcodeproj/project.pbxproj]: This also changed to just a warning.

File: FunctionalTableData.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -657,16 +693,34 @@
...
+				GCC_WARN_UNKNOWN_PRAGMAS = YES;
 				GCC_WARN_UNUSED_FUNCTION = YES;
+				GCC_WARN_UNUSED_LABEL = YES;
+				GCC_WARN_UNUSED_PARAMETER = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
 				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
 				MTL_ENABLE_DEBUG_INFO = NO;
 				SDKROOT = iphoneos;
 				SWIFT_OPTIMIZATION_LEVEL = "-Owholemodule";
+				SWIFT_TREAT_WARNINGS_AS_ERRORS = YES;
```
Comment (krbarnes) [inline FunctionalTableData.xcodeproj/project.pbxproj]: I guess this negates my other comments.
Carry on.

Comment (krbarnes) [reply FunctionalTableData.xcodeproj/project.pbxproj]: ah.. GCC_TREAT_WARNINGS_AS_ERRORS=YES probably catches those... 
Stupid Xcode project files.

Comment (raulriera) [reply FunctionalTableData.xcodeproj/project.pbxproj]: Yeah, I also flipped those... I can revert the changes to the manually error ones just in case tho

Comment (krbarnes) [reply FunctionalTableData.xcodeproj/project.pbxproj]: nah... I think it's fine. This project should never have anything but Swift in it anyway.

**Pull request merged**